File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

const appChat = consumer.subscriptions.create("ChatChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
  },

  received: function(group_chats, group_id) {
    $("#group-comment-index-group_id").html("<%= j(render 'public/groups/comment', comments: group_chats) %>");
  },

  speak: function(chat, group_id, current_member) {
    return this.perform('speak', {chat: chat, group_id: group_id, current_member: current_member});
  }
});

document.addEventListener("turbolinks:load", function() {
  const chatSubmit = document.getElementById("chat-submit");
  chatSubmit.addEventListener("click", function(event) {
    const textArea = document.getElementById("text-area")
    const groupId = document.getElementById("group-data")
    appChat.speak(textArea.value, groupId.dataset.groupId, groupId.dataset.currentMember)
    textArea.value = '';
    event.preventDefault();
  });
})

File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    stream_from "chat_channel"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    @group_chat = GroupChat.new
    @group = Group.find_by(id: data['group_id'])
    @group_chat.group_id = data['group_id']
    @group_chat.member_id = data['current_member']
    @group_chat.group_chat = data['chat']
    @group_chat.save
    @group_chats = @group.group_chats
    ActionCable.server.broadcast('chat_channel', [group_chats: @group_chats, group_id: data['group_id']])
  end
end
